app/controllers/attachments_controller.rb

```diff
@@ -119,4 +121,14 @@ def attachment_params
   def attachment_valid?
     !params["attachment"].nil? && !params["attachment"]["file"].nil?
   end
+
+  def ensure_extension_white_listed
+    document = fetch_document
+    file = params.fetch(:attachment, :file)
+    return if file["file"].blank?
+    return if Attachment.valid_filetype(file)
+
+    flash[:danger] = "Invalid file type: #{File.extname(file['file'].tempfile)}"
```

Comment:
Given my previous comment you could just append this message to the existing invalid attachment error message:
```ruby
flash[:danger] = "Adding an attachment failed. Please make sure you have uploaded an attachment of a permitted file type."
```